src/main/java/ladder/domain/Point.java

```diff
@@ -0,0 +1,81 @@
+package ladder.domain;
+
+import ladder.exception.InvalidPointException;
+import ladder.utils.RandomBooleanGenerator;
+
+import java.util.Objects;
+
+public class Point {
+    private final boolean left;
+    private final boolean current;
+
+    public static Point first(boolean current) {
+        return new Point(false, current);
+    }
+
+    public static Point first() {
+        boolean nextRandom = RandomBooleanGenerator.execute();
+
+        return new Point(false, nextRandom);
+    }
+
+    private Point(boolean left, boolean current) {
+        this.left = left;
+        this.current = current;
+    }
+
+
+    public Point next(boolean current) {
+        if (this.current && current) {
+            throw new InvalidPointException();
+        }
+        return new Point(this.current, current);
+    }
+
+    public Point next() {
+        boolean nextRandom = RandomBooleanGenerator.execute();
+
+        if (this.current == true) {
+            nextRandom = false;
+        }
+
+        return new Point(this.current, nextRandom);
+    }
+
+    public Point last() {
+        return new Point(this.current, false);
+    }
+
+    public Direction movedDirection() {
+        if (left == false && current == true) {
```

Comment:
```suggestion
        if (!left && current) {
```
가 가독이 더 좋을거같은데 어떻게 생각하시나요?

Reply:
저도 그렇게 생각합니다 :)